### PR TITLE
overall: add types hints to the small subset used by avatar

### PR DESCRIPTION
**Pull request by uael**
It's **very** not complete, just a small subset

### CODE REVIEW

File: bumble/device.py [hunk 1]
```diff
@@ -842,6 +863,17 @@ def host_event_handler(function):
 
 # -----------------------------------------------------------------------------
 class Device(CompositeEventEmitter):
+    # im-complete list of fields.
```
Comment (barbibulle) [inline bumble/device.py:866]: did you mean "Incomplete"?

File: bumble/device.py [hunk 2]
```diff
@@ -1600,7 +1638,9 @@ def on_connection_failure(error):
             if transport == BT_LE_TRANSPORT:
                 if connection_parameters_preferences is None:
                     if connection_parameters_preferences is None:
-                        connection_parameters_preferences = {
+                        connection_parameters_preferences: dict[
```
Comment (barbibulle) [inline bumble/device.py:1641]: use `Dict` for python 3.8 compatibility.

File: bumble/sdp.py [hunk 3]
```diff
@@ -183,63 +183,63 @@ def __init__(self, element_type, value, value_size=None):
                 raise ValueError('integer types must have a value size specified')
 
     @staticmethod
-    def nil():
+    def nil() -> 'DataElement':
```
Comment (barbibulle) [inline bumble/sdp.py:186]: (this line and same below): use unquoted class name

**Comment by uael** [comment]
I updated the code to avoid re-format from a newer black :+1:

File: bumble/device.py [hunk 4]
```diff
@@ -511,6 +513,17 @@ class ConnectionParametersPreferences:
 
 # -----------------------------------------------------------------------------
 class Connection(CompositeEventEmitter):
+    device: 'Device'
```
Comment (barbibulle) [inline bumble/device.py:516]: no quotes needed on `Device` class name.

File: bumble/device.py [hunk 5]
```diff
@@ -730,12 +749,12 @@ def __init__(self):
...
         self.keystore = None
-        self.gatt_services = []
+        self.gatt_services: List[Dict[str, Any]] = []
 
-    def load_from_dict(self, config):
+    def load_from_dict(self, config: Dict[str, Any]) -> None:
         # Load simple properties
         self.name = config.get('name', self.name)
-        self.address = Address(config.get('address', self.address))
+        self.address = Address(config.get('address', DEVICE_DEFAULT_ADDRESS))
```
Comment (barbibulle) [inline bumble/device.py:757]: The intent of the original code was to only change address if specified in the dict, else leave it unchanged (i.e don't force it to DEVICE_DEFAULT_ADDRESS).

File: bumble/device.py [hunk 6]
```diff
@@ -2243,7 +2286,9 @@ def on_encryption_failure(error_code):
             )
 
     # [Classic only]
-    async def request_remote_name(self, remote):  # remote: Connection | Address
+    async def request_remote_name(
+        self, remote: Union[Address, Connection]
+    ) -> str:  # remote: Connection | Address
```
Comment (barbibulle) [inline bumble/device.py:2291]: nit: the comment is probably not needed anymore now that there's a type.

**Review by barbibulle — APPROVED** [review]

**Pull request merged**